# The old way of getting the MODULE_VERSION for packaging assumed a particular
# tag format which is only used inside AMD's internal repositories. This
# changes the version checking mechanism to pay attention to github tags.
diff --git a/Makefile b/Makefile
index 9f8960e..e73ee14 100644
--- a/Makefile
+++ b/Makefile
@@ -11,10 +11,12 @@ BUILDDIR = $(BUILD_ROOT)/$(MAKECMDGOALS)
 PACKAGE_DIR = $(BUILD_ROOT)/rocm-smi
 DEBIAN_DIR = $(SMI_ROOT)/DEBIAN
 SMI_LOCATION = $(PACKAGE_DIR)/opt/rocm/bin
-MODULE_VERSION = $(shell cd ${SMI_ROOT} && git describe --long --dirty --match [0-9]*)
+MODULE_VERSION = 1.0.0_$(shell cd ${SMI_ROOT} && git describe --long --tags | sed 's/[a-zA-Z]\+-//g')
+NUMERIC_VERSION = 1.0.0.$(shell cd ${SMI_ROOT} && git describe --tags | sed 's/[a-zA-Z]\+-//g')

 export SMI_ROOT
 export MODULE_VERSION
+export NUMERIC_VERSION

 package-common:
        @mkdir -p $(BUILDDIR)
@@ -22,7 +24,7 @@ package-common:
 deb: package-common
	@mkdir -p $(PACKAGE_DIR)/DEBIAN
	@mkdir -p $(SMI_LOCATION)
-	@sed 's/^Version: MODULE_VERSION/Version: $(MODULE_VERSION)/' $(DEBIAN_DIR)/control > $(PACKAGE_DIR)/DEBIAN/control
+	@sed 's/^Version: MODULE_VERSION/Version: $(NUMERIC_VERSION)/' $(DEBIAN_DIR)/control > $(PACKAGE_DIR)/DEBIAN/control
	@cp $(SMI_ROOT)/rocm_smi.py $(SMI_LOCATION)
	@ln -srf $(SMI_LOCATION)/rocm_smi.py $(SMI_LOCATION)/rocm-smi
	@fakeroot dpkg-deb --build $(PACKAGE_DIR) \
